aten/src/ATen/native/TensorShape.cpp

```diff
@@ -368,7 +368,7 @@ static Tensor cat_sparse(TensorList tensors, int64_t dim) {
     // The dimension in each tensor's values object that corresponds to the overall dimension along which we're catting.
     int64_t values_dim = wrapped - sparse_dim + 1;
     // The final size along the catted dimension.
-    int64_t total_size = std::accumulate(tensors.begin(), tensors.end(), 0, [values_dim](int64_t l, Tensor const &r) {
+    const int64_t total_size = std::accumulate(tensors.begin(), tensors.end(), decltype(total_size){0}, [values_dim](int64_t l, Tensor const &r) {
```

Comment:
It feels neater to make `decltype(total_size){0}` instead `static_cast<int64_t>(0)` - that way if someone switches this to a `const auto` later the meaning doesn't get lost.

Reply:
It also makes the return-holding variable dependent on the value, rather than the other way around.